spec/requests/health_controller_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Health Endpoint', type: :request do
+  describe 'GET #show' do
+    context 'when the application is up' do
+      it 'returns a successful response with HTML' do
+        get '/'
+
+        expect(response).to have_http_status(:ok)
+        expect(response.body).to include('<html><body style="background-color: green"></body></html>')
+      end
+
+      it 'returns a successful response with JSON' do
+        get '/', headers: { 'ACCEPT' => 'application/json' }
+
+        expect(response).to have_http_status(:ok)
+        expect(response.body).to eq({ message: 'OK' }.to_json)
+      end
+    end
+
+    context 'when the application is down' do
+      it 'returns an internal server error response with HTML' do
+        controller_instance = HealthController.new
```

Comment:
Initially, I tried with `controller_instance = instance_double(HealthController)`

But this would result to, not sure if if I'm doing this correctly 💭 

```
Failure/Error: get '/'
#<InstanceDouble(HealthController) (anonymous)> received unexpected message :dispatch with ("show", #<ActionDispatch::Request GET "http://www.example.com/" for 127.0.0.1>, #<ActionDispatch::Response:0x00007f8d072ad5a0 @mon_data=#<Monitor:0x00007f8d072ad2f8>, @mon_data_owne... @cache_control={}, @request=#<ActionDispatch::Request GET "http://www.example.com/" for 127.0.0.1>>)
```

Reply:
Controllers are a bit of a pain in Rails so let's not worry too much about it in this case 😌